packages/story-editor/src/components/videoTrim/videoTrimmer.js

```diff
@@ -55,38 +48,95 @@ function VideoTrimmer() {
     maxOffset,
     setStartOffset,
     setEndOffset,
+    trimBackgroundImage,
+    hasChanged,
+    performTrim,
+    resetOffsets,
   } = useVideoTrim(
     ({
-      state: { currentTime, startOffset, endOffset, maxOffset },
-      actions: { setStartOffset, setEndOffset },
+      state: {
+        currentTime,
+        startOffset,
+        endOffset,
+        maxOffset,
+        trimBackgroundImage,
+        hasChanged,
+      },
+      actions: { setStartOffset, setEndOffset, performTrim, resetOffsets },
     }) => ({
       currentTime,
       startOffset,
       endOffset,
       maxOffset,
       setStartOffset,
       setEndOffset,
+      trimBackgroundImage,
+      hasChanged,
+      performTrim,
+      resetOffsets,
     })
   );
+  const { pageWidth } = useLayout(({ state: { pageWidth } }) => ({
+    pageWidth,
+  }));
+
+  if (!pageWidth || !maxOffset) {
+    return null;
+  }
 
   return (
     <Menu>
-      <DummyLabel>{'Start: '}</DummyLabel>
-      <DummyInput
-        max={maxOffset}
-        value={startOffset}
-        onChange={(evt) => setStartOffset(evt.target.valueAsNumber)}
-      />
-
-      <DummyLabel>{'Current: '}</DummyLabel>
-      <DummyInput disabled value={currentTime} max={maxOffset} />
-
-      <DummyLabel>{'End: '}</DummyLabel>
-      <DummyInput
-        value={endOffset}
-        max={maxOffset}
-        onChange={(evt) => setEndOffset(evt.target.valueAsNumber)}
-      />
+      {hasChanged && (
+        <ButtonWrapper isStart>
+          <Button
+            variant={BUTTON_VARIANTS.RECTANGLE}
+            type={BUTTON_TYPES.SECONDARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={resetOffsets}
+          >
+            {__('Cancel', 'web-stories')}
+          </Button>
+        </ButtonWrapper>
+      )}
+      <Wrapper pageWidth={pageWidth} backgroundImage={trimBackgroundImage}>
+        <UnusedAtStart width={(startOffset / maxOffset) * pageWidth} />
+        <UnusedAtEnd
+          width={((maxOffset - endOffset) / maxOffset) * pageWidth}
+        />
+        <CurrentTime
+          railWidth={pageWidth}
+          aria-label={__('Current time', 'web-stories')}
+          disabled
+          value={currentTime}
+          max={maxOffset}
+        />
+        <Handle
+          railWidth={pageWidth}
+          max={maxOffset}
+          value={startOffset}
+          aria-label={__('Start offset', 'web-stories')}
+          onChange={(val) => setStartOffset(val)}
+        />
+        <Handle
+          railWidth={pageWidth}
+          value={endOffset}
+          max={maxOffset}
+          aria-label={__('End offset', 'web-stories')}
+          onChange={(val) => setEndOffset(val)}
+        />
+      </Wrapper>
+      {hasChanged && (
+        <ButtonWrapper>
+          <Button
+            variant={BUTTON_VARIANTS.RECTANGLE}
+            type={BUTTON_TYPES.PRIMARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={performTrim}
```

Comment:
Q: Currently nothing happens visibly when clicking this button, is it expected not to be functional currently?

Reply:
To be done in #9105